### ISSUE TITLE
A User should be able to purchase tickets using a previously saved payment method

### ISSUE THREAD

**Issue by zmelamed** [big-neon/bn-api#169]
child of big-neon/docs#22

**Comment by Timo614** [comment]
Most of this should be set via #362 as it allows the saving of Stripe Customer data to be reused for a future purchase. Guess we just need to surface the cards via an endpoint that returns some basic data (ie card ending 2222) via https://stripe.com/docs/api#list_cards and then allow the frontend to pass along the selected card to make the purchase. Or if we're not surfacing the a set of cards for them to pick just charges using the default (although seems to make sense to pass that list back with one including a default flag).

Will grab this once the other card is merged unless someone else does beforehand.

**Comment by Timo614** [comment]
I've spent some time with the logic and noticed we only allow one Stripe card right now using whatever is set as the default for the customer. I guess for the MVP that's probably fine so will move forward with this approach and not implement the list of cards for the customer to then pass in when selecting Stripe (ie: multiple card options for that provider). We can circle back with a different card to add that capability in the future as that seems out of scope here now that I've spent some time with this logic. 

So for this going to:

1. When a new token is used and stored the payment method is updated to use that in place of the old (the logic has a uniqueness constraint on the payment method)
2. Add endpoint to return payment methods for use in checkout
3. Allow the payment method to be used in place of the token when the customer has previously saved it
